core/src/main/scala/org/apache/spark/sql/delta/files/TransactionalWrite.scala

```diff
@@ -215,18 +216,39 @@ trait TransactionalWrite extends DeltaLogging { self: OptimisticTransactionImpl
     writeFiles(data, Nil)
   }
 
+  /**
+   * If this is a CDC write, we need to generate the CDC_PARTITION_COL in order to properly
```

Comment:
nit: I suggest writing the docs such that it explains the use of the function rather than the detail of how. for example, what does this function return? what the 2 tuples?